management/log_test.go

```diff
@@ -1,51 +1,34 @@
 package management
 
 import (
+	"fmt"
 	"testing"
 
-	"github.com/auth0/go-auth0"
+	"github.com/stretchr/testify/assert"
 )
 
-func TestLog(t *testing.T) {
-	// Save a log from the listing so we can look it up by ID later
-	var firstLog *Log
+const successfulAPIOperation = "sapi"
 
-	t.Run("List", func(t *testing.T) {
-		// Limit results to 5 entries, starting from the first page
-		logs, err := m.Log.List(Page(1), PerPage(5))
-		if err != nil {
-			t.Error(err)
-		}
+func TestLogManager(t *testing.T) {
+	// Limit results to 5 entries, starting from the first page.
+	logs, err := m.Log.List(Page(1), PerPage(5))
+	assert.NoError(t, err)
+	assert.Greater(t, len(logs), 0, "can't seem to find any logs, have you ran any tests before?")
 
-		for i, log := range logs {
-			t.Logf("%v\n", log)
+	actualLog, err := m.Log.Read(logs[0].GetID())
+	assert.NoError(t, err)
+	assert.Equal(t, logs[0], actualLog)
 
-			// Save the first log for reading later
-			if i == 0 {
-				firstLog = log
-			}
-		}
-	})
+	// Search by type "Success API Operation" and limit results to 5 entries.
+	logs, err = m.Log.List(
+		Parameter("q", fmt.Sprintf(`type:%q`, successfulAPIOperation)),
+		PerPage(5),
+	)
+	assert.NoError(t, err)
 
-	t.Run("Read", func(t *testing.T) {
-		log, err := m.Log.Read(auth0.StringValue(firstLog.ID))
-		if err != nil {
-			t.Error(err)
-		}
-		t.Logf("%v\n", log)
-	})
-
-	t.Run("Search", func(t *testing.T) {
-		// Search by type "Success Exchange" and limit results to 5 entries
-		logs, err := m.Log.List(Parameter("q", `type:"seacft"`), PerPage(5))
-		if err != nil {
-			t.Error(err)
-		}
-		for _, log := range logs {
-			t.Logf("%v\n", log)
-			if auth0.StringValue(log.Type) != "seacft" {
-				t.Error("unexpected log type found")
-			}
+	for _, log := range logs {
+		if log.GetType() != successfulAPIOperation {
+			t.Errorf("unexpected log type found: %s, for log with id: %s", log.GetType(), log.GetID())
 		}
-	})
+	}
```

Comment:
Is this necessary, since giving a correct response is up to the Management API instead of the SDK?

Reply:
Good point, removed this in [351c7d4](https://github.com/auth0/go-auth0/pull/61/commits/351c7d4dfe0abd9859bfccfd3bffdbf3a6465b1d). It was something leftover from before, wasn't too sure either if to leave it or not but makes sense to not have it in there.